### DISCUSSION TITLE
Real-time collaboration

### DISCUSSION THREAD

**Discussion by alokedesai** [warpdotdev/Warp#634] [Product roadmap]
**Is your feature request related to a problem? Please describe.**
The terminal is inherently a single-player experience, which is frustrating if you want to pair with someone in a live session. This can be useful for cases like dev-ops where multiple people could be debugging something in a single session at the same time. 

**Describe the solution you'd like**
I would like to be able to join a session via a link and richly interact with it like it was a session running in my native terminal. I'd also like the session to be optionally persisted after it ends so there can be an audit history of what happened in the session that is easily sharable. 

**Additional context**
Technically we would start by implementing Warp in the browser by compiling Warp to WASM. We'd also need to build the correct architecture (likely through websockets) so that we can broadcast changes from the native app to the browser and vice-versa.

**Comment by elviskahoro** [comment]
From @drmason13 in Discord - https://discord.com/channels/851854972600451112/906269689116770304/906279736492314645

> I love the idea of shared/multiplayer terminals. I wonder who would be running the terminal though. Like, do I start on my machine and share that with someone else? Or are we both connected somehow? Can I run a command from my machine on their filesystem. It seems crazy but might be possible!

**Comment by elviskahoro** [comment]
From @chrispy via Discord - https://discord.com/channels/851854972600451112/906279736492314645/906285296260366379

>I'd love a smooth process here with possibly a cloud-hosted thing or even just local containers on the host's machine. I've always been iffy with delegating remote terminal access to some tool, even if I trust my coworker.

>I could imagine Warp either hosting a collab container themselves to make it very very smooth or spawning one in your own cloud env.

**Comment by Vermyndax** [comment]
I could have *really* benefitted from this feature today, as I was working with someone on a Terraform problem. This would truly be amazing. +1.
